File: app/src/main/java/dev/suhockii/lifetest/di/ActivityCreationListener.java
package dev.suhockii.lifetest.di;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

public abstract class ActivityCreationListener implements Application.ActivityLifecycleCallbacks {
    @Override
    public void onActivityStarted(Activity activity) { /*ignore*/ }

    @Override
    public void onActivityResumed(Activity activity) { /*ignore*/ }

    @Override
    public void onActivityPaused(Activity activity) { /*ignore*/ }

    @Override
    public void onActivityStopped(Activity activity) { /*ignore*/ }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) { /*ignore*/ }

    @Override
    public void onActivityDestroyed(Activity activity) { /*ignore*/ }
}

File: app/src/main/java/dev/suhockii/lifetest/di/ActivityCreationListener.kt
package dev.suhockii.lifetest.di

import android.app.Activity
import android.app.Application
import android.os.Bundle

abstract class ActivityCreationListener : Application.ActivityLifecycleCallbacks {
    override fun onActivityStarted(activity: Activity) { /*ignore*/
    }

    override fun onActivityResumed(activity: Activity) { /*ignore*/
    }

    override fun onActivityPaused(activity: Activity) { /*ignore*/
    }

    override fun onActivityStopped(activity: Activity) { /*ignore*/
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) { /*ignore*/
    }

    override fun onActivityDestroyed(activity: Activity) { /*ignore*/
    }
}
